### ISSUE TITLE
[elixir/en] Clarification needed

### ISSUE THREAD

**Issue by arzeth** [adambard/learnxinyminutes-docs#2594]
```elixir
defmodule Recursion do
  def sum_list([head | tail], acc) do
    sum_list(tail, acc + head)
  end

  def sum_list([], acc) do
    acc
  end
end

Recursion.sum_list([1,2,3], 0) #=> 6
```
I spent 5 minutes thinking why the second ```sum_list``` doesn't redefine the first ```sum_list``` if the first argument in both functions is a list, because I thought that pattern matching in Elixir is just a syntactic sugar as in JS (ES6).

I think there should be a comment like:
```This function is invoked only when the first argument is an empty list```
But I don't know whether I correctly and nicely formulated the comment, so I think that someone else should think and commit.

**Comment by vendethiel** [comment]
It should probably be explained around `defmodule Geometry`.

**Comment by chriszimmerman** [comment]
I opened https://github.com/adambard/learnxinyminutes-docs/pull/3261 to attempt to address your concerns.

**Comment by divayprakash** [comment]
Closed in #3270
